File: Employee.php
<?php
abstract class Employee{
    private string $name;
    private String $employeeId;

    public function __construct(string $name, string $employeeId) {
        $this->name = $name;
        $this->employeeId = $employeeId;
    }

    public function getName():string { 
        return $this->name; 
    }

    public function getEmployeeId(): string {
        return $this->employeeId;
    }

    abstract public function calculatePay(): string;

}

?>

File: employee.py
from abc import ABC, abstractmethod

class Employee(ABC):
    def __init__(self, name, employee_id):
        self._name = name
        self._employee_id = employee_id
        

    
    # Getter methods
    @property
    def name(self):
        return self._name

    @property
    def employee_id(self):
        return self._employee_id

    # Abstract method
    @abstractmethod
    def calculate_pay(self):
        pass